pct.sh

```diff
@@ -18,23 +18,99 @@ if [[ -n ${EXTRA_MAVEN_PROPERTIES-} ]]; then
+	# pre-Jakarta imports) and we do not want it to be upgraded to a version that uses Jakarta
+	# imports (which is not a realistic test scenario) just because the Jakarta Mail API plugin
+	# happens to be on the class path and triggers an upper bounds violation.
+	#
+	# javax.servlet:servlet-api comes from core at version 0, which is an intentional trick to
+	# prevent this library from being used, and we do not want it to be upgraded to a nonzero
+	# version (which is not a realistic test scenario) just because it happens to be on the
+	# class path of some plugin and triggers an upper bounds violation. JENKINS-68696 tracks the
+	# removal of this trick.
+	#
+	echo upperBoundsExcludes=com.sun.mail:jakarta.mail,javax.servlet:servlet-api >"$(pwd)/maven.properties"
+else
+	#
+	# javax.servlet:servlet-api comes from core at version 0, which is an intentional trick to
+	# prevent this library from being used, and we do not want it to be upgraded to a nonzero
+	# version (which is not a realistic test scenario) just because it happens to be on the
+	# class path of some plugin and triggers an upper bounds violation. JENKINS-68696 tracks the
+	# removal of this trick.
+	#
+	echo upperBoundsExcludes=javax.servlet:servlet-api >"$(pwd)/maven.properties"
+fi
+
+#
+# Apache HttpComponents Client 4.x API, Oracle Java SE Development Kit Installer, JSch, OkHttp, and
+# Plain Credentials use an older plugin parent POM and therefore an older test harness that predates
+# compatibility with the removal of JNR in recent cores in jenkinsci/jenkins-test-harness#350. As a
+# temporary workaround, we override the test harness to a recent version. Note that we cannot use a
+# test harness newer than 1812.v6d4e97d91fd8, because later releases of the test harness require
+# changes to the plugin parent POM for JUnit 5 support.
+#
+# TODO When these plugins are using a plugin parent POM with test harness 1657.vf8a824e79147 or
+# later (i.e., plugin parent POM 4.32 or later), this can be deleted.
+#
+if [[ $PLUGINS =~ apache-httpcomponents-client-4-api ]] ||
+	[[ $PLUGINS =~ jdk-tool ]] ||
+	[[ $PLUGINS =~ jsch ]] ||
+	[[ $PLUGINS =~ okhttp-api ]] ||
+	[[ $PLUGINS =~ plain-credentials ]]; then
+	MAVEN_PROPERTIES+=:jenkins-test-harness.version=1812.v6d4e97d91fd8
+fi
+
+#
+# Testing plugins against a version of Jenkins that requires Java 11 exposes
+# jenkinsci/plugin-pom#563. This was fixed in plugin parent POM 4.42, but many plugins under test
+# still use an older plugin parent POM. As a temporary workaround, we skip Enforcer.
+#
+# TODO When all plugins in the managed set are using plugin parent POM 4.42 or later, this can be
+# deleted.
+#
+MAVEN_PROPERTIES+=:enforcer.skip=true
+
 java \
-	--add-opens java.base/java.lang.reflect=ALL-UNNAMED \
-	--add-opens java.base/java.text=ALL-UNNAMED \
-	--add-opens java.base/java.util=ALL-UNNAMED \
-	--add-opens java.desktop/java.awt.font=ALL-UNNAMED \
```

Comment:
No longer needed as of https://github.com/jenkinsci/plugin-compat-tester/pull/352.